optuna/storages/_in_memory.py

```diff
@@ -129,7 +118,9 @@ def get_study_directions(self, study_id: int) -> List[StudyDirection]:
 
         with self._lock:
             self._check_study_id(study_id)
-            return self._studies[study_id].directions
+            directions = self._studies[study_id].directions
+            assert directions is not None  # Study directions are never `None`.
```

Comment:
Do we need this assertion? At least this PR does not seem to increase the possibility of `study.directions` being None here.

Reply:
It's a type-checking thing, `pyright` LSP will cause an issue `directions are possibly None` so I add it. Of course, it's a bit redundant.

Reply:
You can avoid this assertion by re-defining the `_StudyInfo` as follow.
```python
class _StudyInfo:
    def __init__(self, name: str, directions: List[StudyDirection]) -> None:
        self.trials: List[FrozenTrial] = []
        self.param_distribution: Dict[str, distributions.BaseDistribution] = {}
        self.user_attrs: Dict[str, Any] = {}
        self.system_attrs: Dict[str, Any] = {}
        self.name: str = name
        self.directions: List[StudyDirection] = directions
        self.best_trial_id: Optional[int] = None
```